docs/src/pages/components/data-grid/editing/editing.md

```diff
@@ -5,7 +5,129 @@ components: DataGrid, XGrid
+
+- A <kbd class="key">Enter</kbd> keydown
+- A <kbd class="key">Backspace</kbd> or <kbd class="key">Delete</kbd> keydown. It will also delete the value and stops the edit mode instantly.
+- A keydown of any printable key, for instance `a`, `E`, `0`, or `$`
+- A double click on the cell
+- A call to `apiRef.current.setCellMode()`.
+
+```
+ /**
+  * Set the cellMode of a cell.
+  * @param GridRowId
+  * @param string
+  * @param 'edit' | 'view'
+  */
+ setCellMode: (id: GridRowId, field: string, mode: GridCellMode) => void;
+```
+
+{{"demo": "pages/components/data-grid/editing/BasicEditingGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Stop editing
+
+If a cell is in edit mode and has focus, any of the following interactions will stop the edit mode:
+
+- A <kbd class="key">Escape</kbd> keydown. It will also roll back changes done in the value of the cell.
+- A <kbd class="key">Tab</kbd> keydown. It will also save and goes to next cell on the same row.
+- A <kbd class="key">Enter</kbd> keydown. It will also save and goes to next cell on the same column.
+- A mousedown outside the cell
+
+### Control cell editability
+
+In addition to the `editable` flag on columns, control which cell is editable using `isCellEditable` prop.
+
+In this demo, the column `Name` is not editable. Only the rows with an even `Age` value are editable.
+The editable cells have a green background for better visibility.
+
+{{"demo": "pages/components/data-grid/editing/IsCellEditableGrid.js", "bg": "inline"}}
+
+### Control editing
+
+The `editRowsModel` prop lets you control the editing state.
+You can handle the `onEditRowModelChange` callback to control the `GridEditRowsModel` state.
+
+{{"demo": "pages/components/data-grid/editing/EditRowModelControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+
+### Column with valueGetter
+
+You can control the committed value when the edit move stops with the `onEditCellChangeCommitted` prop.
+This is especially interesting when using the `valueGetter` on the column definition.
+
+{{"demo": "pages/components/data-grid/editing/ValueGetterGrid.js", "bg": "inline"}}
+
+### Client-side validation
+
+Follow the following steps to validate the value in the cells.
+
+- Set the event handler `onEditCellChange`. It's invoked when a change is triggered by the edit input component.
+- Set the event handler `onEditCellChangeCommitted` to validate or persist the new value. This handler is invoked when an end-user requests a change to be committed and before the cell reverts to view mode.
+
+Alternatively, you can use the `GridEditRowsModel` state mentioned in the [Control editing](#control-editing) section.
+
+{{"demo": "pages/components/data-grid/editing/ValidateRowModelControlGrid.js", "bg": "inline"}}
+
+#### Using apiRef [<span class="pro"></span>](https://material-ui.com/store/items/material-ui-pro/)
+
+{{"demo": "pages/components/data-grid/editing/ValidateCellApiRefGrid.js", "bg": "inline"}}
```

Comment:
Without context, I had no idea what this demo was about. Looking closer, I understand the following:

```suggestion
You can reproduce the same behavior using the apiRef.

{{"demo": "pages/components/data-grid/editing/ValidateCellApiRefGrid.js", "bg": "inline"}}
```